search/api.py

```diff
@@ -0,0 +1,207 @@
+"""
+Functions for search
+"""
+from itertools import islice
+import logging
+
+from django.conf import settings
+from elasticsearch.helpers import bulk
+from elasticsearch.exceptions import NotFoundError
+from elasticsearch_dsl import Mapping
+from elasticsearch_dsl.connections import connections
+
+from profiles.models import Profile
+from profiles.serializers import ProfileSerializer
+from search.exceptions import ReindexException
+
+log = logging.getLogger(__name__)
+
+
+PROFILE_DOC_TYPE = 'profile'
+
+DOC_TYPES = (PROFILE_DOC_TYPE, )
+
+_CONN = None
+# When we create the connection, check to make sure all appropriate mappings exist
+_CONN_VERIFIED = False
```

Comment:
i'm ok with the usage of `global` in the function(s) below, but i think it would help to encapsulate `_CONN` and `_CONN_VERIFIED` in a simple class, eg: `ESConnection`?  lines 24 & 26 would become something like `es_connection = ESConnection()`. some other possible changes as a result:

- line 38: `if _CONN is None` --> `if not es_connection.created():`
- line 39: `_CONN = connections.create_connection...` --> `es_connection.set(connections.create_connection...)`
- line 43: `if verify and not _CONN_VERIFIED:` --> `if verify and not es_connection.verified():`
- create helper methods for `_CONN.indices.exists...`, `_CONN.indices.get_mapping()...` etc

since `get_conn` is used in a bunch of places, maybe it should still return that object (ie: `return es_connection.conn`)